src/main/java/sirius/pasta/noodle/macros/InlineSvgMacro.java

```diff
@@ -0,0 +1,129 @@
+
+import javax.annotation.Nonnull;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.util.List;
+
+/**
+ * Provides a macro which inlines an SVG file.
+ */
+@Register
+public class InlineSvgMacro extends BasicMacro {
+
+    @Part
+    private Resources resources;
+
+    @Override
+    protected Class<?> getType() {
+        return String.class;
+    }
+
+    @Override
+    protected void verifyArguments(CompilationContext compilationContext, Position position, List<Class<?>> args) {
+        if (args.size() != 1 || !CompilationContext.isAssignableTo(args.get(0), String.class)) {
+            throw new IllegalArgumentException("Expected a single String as argument.");
+        }
+    }
+
+    @Override
+    public void verify(CompilationContext context, Position position, List<Node> args) {
+        super.verify(context, position, args);
+
+        if (args.get(0).isConstant()) {
+            String resourceName = String.valueOf(args.get(0).getConstantValue());
+            if (resources.resolve(resourceName).isEmpty()) {
+                context.warning(position, "Unknown resource: %s", resourceName);
+            }
+        }
+    }
+
+    @Override
+    public boolean isConstant(CompilationContext context, List<Node> args) {
+        return args.get(0).isConstant();
+    }
+
+    @Override
+    public Object invoke(Environment environment, Object[] args) {
+        String path = (String) args[0];
+        return processResource(path);
+    }
+
+    @Nonnull
+    private String processResource(String path) {
+        if (!path.startsWith("/assets")) {
+            throw new IllegalArgumentException("Only assets can be inlined for security reasons.");
+        }
+
+        Resource resource = resources.resolve(path).orElse(null);
+        if (resource == null) {
+            return "";
+        }
+
+        Element root = parseDocument(resource).getDocumentElement();
+        if (!Strings.areEqual("svg", root.getTagName())) {
+            throw Exceptions.createHandled().withDirectMessage("The referenced resource is not an SVG file.").handle();
+        }
+
+        return stringifyElement(root);
+    }
+
+    private Document parseDocument(Resource resource) {
+        try (InputStream stream = resource.openStream()) {
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            return factory.newDocumentBuilder().parse(stream);
+        } catch (Exception exception) {
+            throw Exceptions.handle(exception);
+        }
+    }
+
+    private String stringifyElement(Element element) {
+        try (StringWriter writer = new StringWriter()) {
+            TransformerFactory factory = TransformerFactory.newInstance();
```

Comment:
Can we strip this variable and just use `TransformerFactory.newInstance().newTransformer()` below? Or maybe even better, lazy-load it one inside the part?

Reply:
This is pretty much the standard approach we use elsewhere, too, for instance when parsing a SAML response. I am not sure whether it is worth the effort to optimise away this single variable, considering that we are loading files from disk and sending them away over the network. (And I would expect the JIT to do that automatically anyway?)

Reply:
But each call will create a new factory, that could be done once as long as the part lives.
Something like that would improve it a bit, especially if this macro is called lots of times?
```
TransformerFactory factory = null;
function fetchTransformerFactory() {
  if (factory == null) {
    this.factory = TransformerFactory.newInstance();
  }
  return factory;
}
```

Reply:
My 2 ct: Most of the time the argument passed to this macro will be constant (as in a string with the path to the SVG) in which case I would guess Tagliatelle can optimize caching and wont invoke the macro all of the time in those places.

Reply:
I understand the concerns, but given the caching/constant nature of most calls, I'd go with the current implementation in the name of simplicity ;-)